### DISCUSSION TITLE
How to install next-js with react 17?

### DISCUSSION THREAD

**Discussion by mxcdh** [vercel/next.js#36006] [Help] [answered]
Hello

How I can install next.js with React 17? When I type:
`npx create-next-app --ts`, npx installed Next with React 18, but I can't use React 18, because Stripe not working at the moment with R18 (https://www.npmjs.com/package/stripe).

**Comment by icyJoseph** [comment, marked as answer]
After you run your command do:

```shell
npm install next@12.1.4 react@17.0.2 react-dom@17.0.2 --exact
```

and then

```shell
npm install --save-dev @types/react@17.0.39 --exact
```

And you should be fine.

**Comment by Mihai-github** [comment]
and what is the version for react dom types?

this one? `"@types/react-dom": "17.0.2",`
